core/templates/domain/collection/collection-node.model.ts

```diff
@@ -60,35 +83,36 @@ export class CollectionNode {
   // This property is immutable. The value returned by this function is
   // null if doesExplorationExist() returns false.
   getExplorationTitle(): string {
-    if (this._explorationSummaryObject) {
+    if (this.doesExplorationExist()) {
       return this._explorationSummaryObject.title;
     } else {
-      return null;
+      return '';
     }
   }
 
   // Returns whether the exploration referenced by this node is known to exist
   // in the backend. This property is immutable.
   doesExplorationExist(): boolean {
-    return this._explorationSummaryObject !== null;
+    return this._explorationSummaryObject.category !== '';
```

Comment:
Hm is this correct? I don't understand why checking the category field signals if the exploration exists.

Reply:
Previously the explorationSummaryObject was initialized with a null value ,this was giving error on running strict_checks so I initialized it by an empty object(see `exploration_summary`).Since I initialized it with an empty but not null value,the logic here got reversed.So to keep the functionalities as they were before I had to check for something that should return same boolean values when checking this condition.I'm not sure if property category is correct here, We should place some property which is mandatory in `explorationSummaryObject` and which can't be empty when we first call `createFromExplorationId`

Reply:
Also for the function `createFromExplorationId` if we only pass an `explorationId` then the previous approach states to always return false.This functionality will remain as it is even if we check any of the properties from `explorationSummaryObject`

Reply:
But an exploration can exist with an empty category right?

Reply:
I don't think so ,for reference see line `71-79` of `collection-node.model.spec.ts`

Reply:
I don't think this is the correct approach here. If you want to refactor this properly you would need to modify the behaviour from the callsites.

Basically, the fix for `null` should be either adding the type as allowed or refactor it in the way that the `null` is not needed at all. Replacing the `null` with empty string or zero doesn't really help here.

Reply:
The correct type to `exploration_summary` is  `LearnerExplorationSummaryBackendDict` which was already assigned ,it needs to have some value in order to pass it to `create` method ,else it would give `Type 'null' is not assignable to type 'LearnerExplorationSummaryBackendDict'`

Reply:
As I said, firstly check if it is possible to refactor the code in a way that the `exploration_summary` can be filled when creating this model, if it is not possible then add `null` for it.

Reply:
I filled it with empty values and ` | null` can't be assigned to `exploration_summary`,the only thing left is how to check for this condition here.Each of the property in `exploration_summary` is important to be assigned (We can also see this on settings tab while creating an exploration) thus we can check for any of the values among `exploration_summary`.If any of them is assigned empty the exploration fails to exist.A more good property would be title. Title can't be empty therefore exploration would not exist.

Reply:
The question was more like: Can we refactor this in a way that when we create this we can immediately fill the `exploration_summary` with the correct values? If not I think that allowing `null` is fine. Your changes would actually introduce more ambiguity since if I wanted to know if any of the values are set I would need to check if the `title` (or `category`) is set.

Reply:
undid this ,made the type to be `LearnerExplorationSummaryBackendDict | null`